llvm/lib/Target/DirectX/DXILWriter/DXILBitcodeWriter.cpp

```diff
@@ -759,10 +759,6 @@ uint64_t DXILBitcodeWriter::getOptimizationFlags(const Value *V) {
       Flags |= bitc::NoSignedZeros;
     if (FPMO->hasAllowReciprocal())
       Flags |= bitc::AllowReciprocal;
-    if (FPMO->hasAllowContract())
-      Flags |= bitc::AllowContract;
-    if (FPMO->hasApproxFunc())
```

Comment:
Why is "approx func" removed?

Reply:
https://releases.llvm.org/3.7.1/docs/LangRef.html#fast-math-flags 
Not listed in fast math flags for 3.7, is this an error?

Reply:
Ah, good point... I don't want to stand in the way of this getting in, but neither is "AllowContract". Should we be asserting on these instead? (it seems "UnsafeAlgebra" was just renamed to "AllowReassoc")
https://github.com/microsoft/DirectXShaderCompiler/blob/070d0d5a2beacef9eeb51037a9b04665716fd6f3/include/llvm/IR/Operator.h#L171

Reply:
@llvm-beanz thoughts on this?

Reply:
ApproxFunc has no equivalent in LLVM 3.7 so there is no bitcode representation of it. We have no choice but to drop it.

Reply:
Also, in LLVM 3.7 UnsafeAlgebra was a magic catch all flag for "all things fast", it was separated out into granular flags in this commit - https://github.com/llvm/llvm-project/commit/629c41153876b.

It had the impact in the LLVM 3.7 optimizer of controlling both contraction and reassociation (to the extent that LLVM 3.7 had optimizations).